modules/@apostrophecms/doc/index.js

```diff
@@ -1843,8 +1845,130 @@ module.exports = {
         return existing?.aposDocId || self.apos.util.generateId();
       },
 
-      ...require('./lib/legacy-migrations')(self),
-      ...require('./lib/migrations')(self)
+      async getAposDocId({
+        _id, slug, locale
+      }) {
+        if (!_id && !slug) {
+          throw self.apos.error('invalid', 'Either _id or slug must be provided');
+        }
+        if (!locale) {
+          throw self.apos.error('invalid', 'Missing locale');
+        }
+
+        // const req = self.apos.task.getReq({ mode: 'draft', locale });
+        const aposLocale = new RegExp(`^${self.apos.util.regExpQuote(locale)}:`);
+        const criteria = _id
```

Comment:
One last thing, sorry... since the _id parameter has to be a full _id, which already contains the locale, there's no need to make locale mandatory or even part of the query when _id is present.

(Technically we don't need to make a query at all when _id is the argument, because _id already contains aposDocId, but I like the idea of making the query because it validates that the document actually does exist.)